pkg/controller/cmd/options.go

```diff
@@ -138,6 +143,8 @@ type ManagerOptions struct {
 	LeaderElectionID string
 	// LeaderElectionNamespace is the namespace to do leader election in.
 	LeaderElectionNamespace string
+	// Disable is a comma-separated list of controllers and webhooks that should be disabled.
+	Disable string
```

Comment:
`ManagerOptions` reflect all the parameters that are related to building up a `Manager`. `Disable` doesn't correspond to one and thus I'd recommend having this as a separate flag that is a `bool`.

Reply:
I can put it in a separate struct but it needs to be a string, since it's "a comma-separated list of controllers and webhooks that should be disabled"
The idea is to use `--disable=controlplane-controller,controlplane-webhook` in the command line.

Reply:
Then I'd recommend having another `Option` that takes this 'list' and parses it into individual boolean flags, i.e.
```go
type ControllerSwitches struct {
  DisableControlplaneController bool
  DisableControlplaneWebhook bool
}

type ControllerSwitchOptions struct {
  Disable string
  completed *ControllerSwitches
}
```
The `ControllerSwitchOptions` would then have the `AddFlags` as all the other options and can do meaningful parsing and once `Completed` return the parsed `ControllerSwitches` that can then be used to decide whether to add something to the `Manager` or not.

Reply:
This approach has the disadvantage that adding a new controller means adding a new switch, something that's unnecessary and should be avoided IMO. Furthermore, if it can be done this way, this doesn't help with the "builder" discussion, see other comments.

Reply:
It's true that this has the disadvantage of adding more named variables, but the `ControllerSwitches` map could be improved to be initialized with a `map[string]func(manager.Manager) error` that maps from the controller's name to the `AddToManager` function of that particular controller. Then it can do proper flag parsing and even validate when there is a name in that list that does not exist (and fail early) and then build up an `AddToManager` function with the required controllers.